packages/tags/src/Tag.ts

```diff
@@ -71,30 +71,49 @@ export class Tag extends SizedMixin(SpectrumElement, {
     };
 
     private handleKeydown = (event: KeyboardEvent): void => {
-        if (!this.deletable) {
+        if (!this.deletable || this.disabled) {
             return;
         }
         const { code } = event;
+
         switch (code) {
             case 'Backspace':
             case 'Space':
             case 'Delete':
                 this.delete();
-                return;
             default:
                 return;
         }
     };
 
+    private shiftFocusOnEvent(): void {
+        const event = new CustomEvent('sp-tag-focus', {
+            bubbles: true,
+            composed: true,
+            cancelable: true,
+        });
+        this.dispatchEvent(event);
+    }
+
     private delete(): void {
         if (this.readonly) {
             return;
         }
-        this.dispatchEvent(
-            new Event('delete', {
-                bubbles: true,
-            })
-        );
+        this.shiftFocusOnEvent();
```

Comment:
_If_ the right decision is to do this imperatively, shouldn't it go after the remove? Or at least after default isn't prevented?